src/System.Collections/src/System/Collections/Generic/LinkedList.cs

```diff
@@ -188,7 +188,7 @@ public void CopyTo(T[] array, int index)
 
             if (index < 0 || index > array.Length)
```

Comment:
(But yes, it overflows to something in the range `int.MaxValue + 1` to `uint.MaxValue` so as long as the other comparand is guaranteed to be in the range `0` to `int.MaxValue` it gives the same result).